### PR TITLE
WTrialBalance ORG_ID should not be mandatory by default

### LINKED ISSUE #1688: WTrialBalance ORG_ID should not be mandatory by default
Currently the field ORG_ID is mandatory, which is correct if you work with balancing Organizations. But if you use organizations not balancing the field should not be mandatory.
Solution:  class org.adempiere.webui.apps.formWTrialBalance line 323
```
  MAcctSchema acctSchema = MAcctSchema.getClientAcctSchema(Env.getCtx(), Env.getAD_Client_ID(Env.getCtx()))[0];
        if (acctSchema == null)
            FDialog.error(form.getWindowNo(), "@No Acctschema@");
        MAcctSchemaElement acctElementOrg = acctSchema.getAcctSchemaElement(MAcctSchemaElement.ELEMENTTYPE_Organization);
        Boolean orgMandatory = acctElementOrg.isBalanced()?true:false;

        if (pa_ReportCube_ID == 0 || c_PeriodTo_ID == 0 || (orgMandatory && m_AD_Org_ID == 0))
        {
            FDialog.error(form.getWindowNo(), "@Please fill mandatory fields (Cube, Period, Organization).@");
            return;}
```

### PR DESCRIPTION
**Pull request by SusanneCalderon**
Fixes #1688

SpecialForm, Organization only mandatory if defined as balanced.

### CODE REVIEW

**Comment by wdtz** [comment]
Hello @SusanneCalderon , I think it convenient to make a change, the definition of the following points of interest for functional tests:

Benefit of Change
Previous operation procedure
New operation procedure (Change)
In this way, it will be easier to understand the proposed change, thus giving an opportunity to give an opinion and evaluate its optimal functioning.

**Comment by SusanneCalderon** [comment]
Hi @wdtz . You are right.
**Class: Special form WTrialBalance.** This special form shows the accumumated accountings with the filters organization, account from to, period. 
**Benefit of change:**
You can show the accountings of several organizations.
**Previous operation procedure:**
The filter _organization_ is mandatory. This make sense if the accounting configuration for the dimension _organization_ is "balanced". 
New operation procedure:
When the accounting dimension _organization_ is balanced the filter organization is mandatory, otherwise not.

File: zkwebui/WEB-INF/src/org/adempiere/webui/apps/form/WTrialBalance.java [hunk 1]
```diff
@@ -323,11 +319,20 @@ private void loadTable()
...
+
+		//pa_ReportCube_ID = 1000000;
+
+		MAcctSchema acctSchema = MAcctSchema.getClientAcctSchema(Env.getCtx(), Env.getAD_Client_ID(Env.getCtx()))[0];
+		if (acctSchema == null)
+			FDialog.error(form.getWindowNo(), "@No Acctschema@");
+		MAcctSchemaElement acctElementOrg = acctSchema.getAcctSchemaElement(MAcctSchemaElement.ELEMENTTYPE_Organization);
+		Boolean orgMandatory = acctElementOrg.isBalanced()?true:false;
+
+		if (pa_ReportCube_ID == 0 || c_PeriodTo_ID == 0 || (orgMandatory && m_AD_Org_ID == 0)){
```
Comment (yamelsenih) [inline zkwebui/WEB-INF/src/org/adempiere/webui/apps/form/WTrialBalance.java]: Hi @SusanneCalderon i see it but i think that is good practice if you use AD message for show error, maybe you can change it line for:
<pre>
		StringBuffer errorMessage = new StringBuffer();
		if (pa_ReportCube_ID == 0) {
			errorMessage.append("@PA_ReportCube_ID@");
		}
		if (c_PeriodTo_ID == 0) {
			if(errorMessage.length() > 0) {
				errorMessage.append(Env.NL);
			}
			errorMessage.append("@C_PeriodTo_ID@");
		}
		if (orgMandatory && m_AD_Org_ID == 0) {
			if(errorMessage.length() > 0) {
				errorMessage.append(Env.NL);
			}
			errorMessage.append("@AD_Org_ID@");
		}
		if(errorMessage.length() > 0) {
			FDialog.error(form.getWindowNo(), "FillMandatory", errorMessage.toString());
			return;
		}
</>

File: zkwebui/WEB-INF/src/org/adempiere/webui/apps/form/WTrialBalance.java [hunk 2]
```diff
@@ -693,10 +698,18 @@ private void setPeriodToStyle()
 	 * Set style on organization Label
 	 */
 	private void setOrgStyle()
-	{
-		if (fieldOrg.getValue() == null)
+	{MAcctSchema acctSchema = MAcctSchema.getClientAcctSchema(Env.getCtx(), Env.getAD_Client_ID(Env.getCtx()))[0];
+		if (acctSchema == null)
+			FDialog.error(form.getWindowNo(), "@No Acctschema@");
```
Comment (yamelsenih) [inline zkwebui/WEB-INF/src/org/adempiere/webui/apps/form/WTrialBalance.java]: Please change **@NoAccrschema@** for **@C_AcctSchema_ID@ @NotFound@**

File: zkwebui/WEB-INF/src/org/adempiere/webui/apps/form/WTrialBalance.java (hunk 1 above)
Comment (SusanneCalderon) [reply zkwebui/WEB-INF/src/org/adempiere/webui/apps/form/WTrialBalance.java]: Hi @yamelsenih 
You are right, I will change it

Comment (SusanneCalderon) [reply zkwebui/WEB-INF/src/org/adempiere/webui/apps/form/WTrialBalance.java]: Hi @yamelsenih ; I included your suggestion

**Comment by yamelsenih** [comment]
Please resolve conflict

**Comment by yamelsenih** [comment]
Any update?

**Comment by SusanneCalderon** [comment]
Hi @yamelsenih 
I replaced the old version of adempiere.iml with the new version from @homebeaver

**Pull request closed without merging**